app/src/main/java/quizapp/volkova/weatherapp/MainActivity.kt

```diff
@@ -1,11 +1,142 @@
 package quizapp.volkova.weatherapp
 
-import androidx.appcompat.app.AppCompatActivity
+import android.R.attr
+import android.animation.ObjectAnimator
+import android.os.AsyncTask
 import android.os.Bundle
+import android.transition.TransitionManager
+import android.util.Log
+import android.view.View
+import android.view.animation.Animation
+import android.view.animation.AnimationUtils
+import android.widget.*
+import androidx.appcompat.app.AppCompatActivity
+import org.json.JSONObject
+import java.net.URL
+import java.text.SimpleDateFormat
+import java.util.*
+import android.R.attr.visible
+
+
+
 
 class MainActivity : AppCompatActivity() {
+    val CITY: String = "Kyiv"
+    val API: String = "407711ecf9fdb73061869dc807029279"
+
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         setContentView(R.layout.activity_main)
+        findWeather().execute()
+    }
+
+    inner class findWeather() : AsyncTask<String, Void, String>() {
```

Comment:
AsyncTask is not an issue at the moment.
Of course, it can be replaced with coroutines, but they will be included in the next blocks (after UI).

Here is the coroutine codelab, if interested:
https://developer.android.com/codelabs/kotlin-coroutines